app/com/gu/floodgate/contentsource/ContentSourceApi.scala

```diff
@@ -66,12 +70,27 @@ class ContentSourceApi(contentSourceService: ContentSourceService, reindexServic
     }
   }
 
+  def cancelReindex(id: String) = Action.async { implicit request =>
+    reindexService.cancelReindex(id) map { happyOrError =>
+      happyOrError match {
+        case Good(_) => Ok
+        case Bad(error) => BadRequest(Json.toJson(ErrorResponse(error.message)))
```

Comment:
Hmmm. There are three error outcomes here:

1. The request to cancel the reindex from the content source tool fails and therefore this is an internal server error.
2. No content source exists for the given content source and therefore this is a bad request.
3. There is no running reindex for this content source and therefore this is a bad request.

I think CustomError type impedes here. Maybe I can group those specific errors under a new type, CancelReindexErrors and match on that to resolve the error response more specifically.